tests/test_ucc.py

```diff
@@ -145,6 +146,51 @@ def test_ucc_circuit(excitation, mapping, basis_rotations, coeffs):
     assert len(test_circuit.get_parameters()) == len(basis_rotations)
 
 
+@pytest.mark.parametrize(
+    "excitation,mapping,basis_rotations,coeffs",
+    [
+        ([0, 2], None, ("Y0 X2", "X0 Y2"), (0.5, -0.5)),  # JW singles
+        (
+            [0, 1, 2, 3],
+            None,
+            (
+                "X0 X1 Y2 X3",
+                "Y0 Y1 Y2 X3",
+                "Y0 X1 X2 X3",
+                "X0 Y1 X2 X3",
+                "Y0 X1 Y2 Y3",
+                "X0 Y1 Y2 Y3",
+                "X0 X1 X2 Y3",
+                "Y0 Y1 X2 Y3",
+            ),
+            (-0.25, 0.25, 0.25, 0.25, -0.25, -0.25, -0.25, 0.25),
+        ),  # JW doubles
+    ],
+)
+def test_qeb_circuit(excitation, mapping, basis_rotations, coeffs):
+    """Build QEB circuit"""
+    theta = 0.1
+    n_qubits = 4
+
+    # Build the control array using SymbolicHamiltonian.circuit
+    # But need to multiply theta by some coefficient introduced by the fermion->qubit mapping
+    control_circuit = Circuit(n_qubits)
+    for coeff, basis_rotation in zip(coeffs, basis_rotations):
+        n_terms = len(basis_rotation)
+        pauli_term = SymbolicHamiltonian(
+            symbols.I(n_qubits - 1)
+            * reduce(lambda x, y: x * y, (getattr(symbols, _op)(int(qubit)) for _op, qubit in basis_rotation.split()))
+        )
+        control_circuit += pauli_term.circuit(-coeff * theta)
+    control_result = control_circuit(nshots=1)
+    control_state = control_result.state(True)
+
+    test_circuit = qeb_circuit(n_qubits, excitation, theta=theta, trotter_steps=1)
```

Comment:
Remove `trotter_steps` argument if not needed

```suggestion
    test_circuit = qeb_circuit(n_qubits, excitation, theta=theta)
```

Reply:
As above. https://github.com/qiboteam/qibochem/pull/113#discussion_r1831894579